kie-internal/src/main/java/org/kie/internal/runtime/manager/audit/query/AuditDeleteBuilder.java

```diff
@@ -55,6 +55,13 @@
      */
     T logBelongsToProcessInDeployment(String deploymentId);
 
+    /**
+     * Specify the number of records to be included per transaction.
+     * @param number of records (0 means all records will be processed into one single transaction)
+     * @return The current query builder instance
+     */
+    T recordsPerTransaction(int numRecords);
```

Comment:
In this particular case (and only in this particular case) since Im pretty certain there is only one implementation (I do not understand why this was an interface in the first place) and the default implementtion will be pretty ugly (a cast to T from this), I prefer to not use this approach